### ISSUE TITLE
flow-for-vscode does not resolve symlinks like flow does

### ISSUE THREAD

**Issue by vicapow** [flowtype/flow-for-vscode#349]
Steps to reproduce:
````sh
mkdir symlinks-bug
cd symlinks-bug
echo '// @flow
export default function foo(): number {
  return "wow";
}' > top-level-original.js
mkdir project1

cd project1

echo '// @flow
import foo from "./symlinked-file.js";
export default function() {
  return foo();
}' > index.js

ln -s ../top-level-original.js symlinked-file.js

yarn init --yes
yarn add flow-bin@latest
yarn flow init
yarn flow
````

Correctly show error from top-level-original

````
Error ../top-level-original.js:3:10
Cannot return "wow" because string [1] is incompatible with number [2].
     1│ // @flow
 [2] 2│ export default function foo(): number {
 [1] 3│   return "wow";
     4│ }
     5│
````

Now open the project in code from project1 directory and index.js from project1/index.js, use command click to jump to definition.

<img width="470" alt="Screen Shot 2019-06-27 at 10 08 26 AM" src="https://user-images.githubusercontent.com/583385/60285961-9ce7ea80-98c3-11e9-84da-8c424b8b86c4.png">

The Flow LSP server correctly takes the user to `top-level-original.js` but this file has no Flow server running for it. This is a little hard to tell because the LSP errors still show up in the file but no type hints or coverage is reported.

<img width="988" alt="Screen Shot 2019-06-27 at 10 10 18 AM" src="https://user-images.githubusercontent.com/583385/60286273-4038ff80-98c4-11e9-9ff7-a3c46039316b.png">

**Comment by Mayank1791989** [comment]
@vicapow Looking only at `top-level-original.js` How will we decide which .flowconfig to use. It's possible we can symlink this file (`top-level-original.js`) into multiple projects say project2, project3 .... and all have different flowconfigs. I am suprised why flow is pointing to original file not symlinked file.

**Comment by vicapow** [comment]
@Mayank1791989 I don't think flow-for-vscode can rely on a 1:1 mapping between file and `.flowconfig.` Imagine a nested project one inside another and a single file would have multiple flowconfigs. and produce different errors depending on lint configurations or flow versions.

**Comment by Mayank1791989** [comment]
> Imagine a nested project one inside another and a single file would have multiple flowconfigs

If we not tie file to single .flowconfig will it not become very confusing to show result from all flow server which this file can be part of. Anyways that is a separate issue.

In the above issue looking at only `top-level-original.js` how will we know in which project file `top-level-original.js` is symlinked and to use that project server to provide all ide features.

**Comment by vicapow** [comment]
I agree it would be confusing to show errors from multiple `.flowconfigs`. I also don't see how it would be possible to know all the flowconfigs that could be associated with that project.

Can we add some sort of workspace configuration? If you open `code` from the root of some project with a flowconfig, use that config for all open files?

> Is type-hint working inside symlinked-file.js?
I believe vscode also resolves the symlink to the original file so there isn't really a symlinked-file.js to type check.

**Comment by Mayank1791989** [comment]
I tried running flow commands directly on symlinked and original file and flow is not working. I think flow only handle errors for symlinked files.

**Comment by vicapow** [comment]
Could you post the commands you ran so I can try to reproduce?

**Comment by Mayank1791989** [comment]
Try running `flow get-def path-to/symlinked-file.js 2 4`

**Comment by vicapow** [comment]
I believe that is a separate bug in flow get-def which tries to search for a flowconfig similar to what flow-for-vscode is going.

**Comment by Mayank1791989** [comment]
Try running any flow command it will behave similarly.

**Comment by Mayank1791989** [comment]
@vicapow lsp is working fine inside symlinked-file.js try opening symlinked-file in vscode and test  hover or go-to-def it will work but all paths which flow server is returning is pointing to original file which I think is wrong it should point to symlinked files not the original.

**Comment by Mayank1791989** [comment]
> Can we add some sort of workspace configuration? If you open code from the root of some project with a flowconfig, use that config for all open files?

Like .flowconfig override for all files in that root?

**Comment by vicapow** [comment]
> Try running any flow command it will behave similarly.
Oh, you're wrong the wrong flow server. you'll need to run the flow server from the root of `project1`.

> Like .flowconfig override for all files in that root?
Not just in that root, any files you open from that vscode workspace should attempt to use that flowconfig. I believe this would match flows behavior.
